parlai/core/torch_ranker_agent.py

```diff
@@ -924,8 +944,8 @@ def _make_candidate_encs(self, vecs):
         self.model.eval()
         with torch.no_grad():
             for vec_batch in tqdm(vec_batches):
-                cand_encs.append(self.encode_candidates(vec_batch))
-        return torch.cat(cand_encs, 0)
+                cand_encs.append(self.encode_candidates(vec_batch).cpu())
```

Comment:
I'm not sure we want this. What led you to adding it?

Thoughts from @emilydinan, @dexterju?

Reply:
it was suggested fix from @klshuster  -- currently it oom without this. it seems superior to me

Reply:
Hmm this will be slower due to transfer but it does have merit.

Reply:
yeah there's a speed tradeoff but if we're encoding too many cands i think it might be necessary